zerver/tests/test_subs.py

```diff
@@ -3125,3 +3125,35 @@ def test_access_stream(self) -> None:
         self.common_subscribe_to_streams(sipbtest.email, [mit_stream.name], subdomain="zephyr")
         access_stream_by_id(sipbtest, mit_stream.id)
         access_stream_by_name(sipbtest, mit_stream.name)
+
+    def test_stream_access_by_guest(self) -> None:
+        polonius = self.example_user('polonius')
+        polonius.is_guest = True
+        polonius.save(update_fields=['is_guest'])
```

Comment:
Same comment on setup.  I'd also call the variable `guest_user_profile`, not `polonius`, for better readability for what's going on.